omnigibson/envs/env_base.py

```diff
@@ -244,15 +264,10 @@ def _load_robots(self):
                     cls_type_descriptor="robot",
                 )
                 # Import the robot into the simulator
-                og.sim.import_object(robot)
-                robot.set_position_orientation(position=position, orientation=orientation)
-
-            if len(self.robots_config) > 0:
-                # Auto-initialize all robots
-                og.sim.play()
-                self.scene.reset()
-                self.scene.update_initial_state()
-                og.sim.stop()
+                self.scene.add_object(robot)
+                robot.set_local_pose(position=position, orientation=orientation)
+                if robot._dummy is not None:
+                    robot._dummy.load(self.scene)
```

Comment:
hmmmm shouldn't the robot class already handle this on its own? seems weird for an external class to access this private var

Reply:
Agreed. @hang-yin please check if there is any reason this has to happen here